prow/plugins/verify-owners/verify-owners.go

```diff
@@ -0,0 +1,151 @@
+	return &pluginhelp.PluginHelp{
+			Description: "The owners plugin validates OWNERS files if they are modified in a PR. On failure it automatically adds a 'do-not-merge/invalid-OWNERS' label to the PR, and a review comment on the incriminated file.",
+		},
+		nil
+}
+
+type ownersClient interface {
+	LoadRepoOwners(org, repo, base string) (repoowners.RepoOwnerInterface, error)
+}
+
+type githubClient interface {
+	AddLabel(org, repo string, number int, label string) error
+	CreateReview(org, repo string, number int, r github.DraftReview) error
+	GetPullRequestChanges(org, repo string, number int) ([]github.PullRequestChange, error)
+}
+
+func handlePullRequest(pc plugins.PluginClient, pre github.PullRequestEvent) error {
+	if pre.Action != github.PullRequestActionOpened && pre.Action != github.PullRequestActionReopened && pre.Action != github.PullRequestActionSynchronize {
+		return nil
+	}
+	return handle(pc.GitHubClient, pc.GitClient, pc.Logger, &pre, pc.PluginConfig.Owners.LabelsBlackList)
+}
+
+func handle(ghc githubClient, gc *git.Client, log *logrus.Entry, pre *github.PullRequestEvent, labelsBlackList []string) error {
+	org := pre.Repo.Owner.Login
+	repo := pre.Repo.Name
+	wrongOwnersFiles := map[string]error{}
+
+	// Get changes.
+	changes, err := ghc.GetPullRequestChanges(org, repo, pre.Number)
+	if err != nil {
+		return fmt.Errorf("error getting PR changes: %v", err)
+	}
+
+	// List modified OWNERS files.
+	var modifiedOwnersFiles []string
+	for _, change := range changes {
+		if filepath.Base(change.Filename) == ownersFileName {
+			modifiedOwnersFiles = append(modifiedOwnersFiles, change.Filename)
+		}
+	}
+	if len(modifiedOwnersFiles) == 0 {
+		return nil
+	}
+
+	// Clone the repo, checkout the PR.
+	r, err := gc.Clone(pre.Repo.FullName)
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := r.Clean(); err != nil {
+			log.WithError(err).Error("Error cleaning up repo.")
+		}
+	}()
+	if err := r.CheckoutPullRequest(pre.Number); err != nil {
+		return err
+	}
+
+	// Check each OWNERS file.
+	for _, f := range modifiedOwnersFiles {
+		// Try to load OWNERS file.
+		simple, err := repoowners.ParseSimpleConfig(filepath.Join(r.Dir, f))
+		if err != nil {
+			wrongOwnersFiles[f] = fmt.Errorf("error occurred parsing the OWNERS file in %s: %v", f, err)
+			break
+		}
+		// Check labels against blacklist
+		if sets.NewString(simple.Config.Labels...).HasAny(labelsBlackList...) {
+			wrongOwnersFiles[f] = fmt.Errorf("OWNERS file contains blacklisted labels: %v", sets.NewString(simple.Config.Labels...).Intersection(sets.NewString(labelsBlackList...)))
+			break
+		}
+		// Check approvers isn't empty
+		if filepath.Dir(f) == "." && len(simple.Config.Approvers) == 0 {
+			wrongOwnersFiles[f] = errors.New("no approvers defined for the repo")
+			break
+		}
+	}
+	// React if we saw something.
+	if len(wrongOwnersFiles) > 0 {
+		log.Debugf("Adding %s label.", invalidOwnersLabel)
```

Comment:
This isn't necessary to log because the GitHub client logs it.
The log for creating the review seems appropriate though since it includes the error count.